### ISSUE TITLE
More detailed and complete example of the main menu screen

### ISSUE THREAD

**Issue by Kenoshen** [bitDecayGames/haxeflixel-template#78]
I would love if it was super easy to customize the main menu with a different title image, button images, etc.

**Comment by MondayHopscotch** [comment]
Yeah, I think we should build a better pattern for this in general. We were using the flixel-ui stuff, but it kinda sucks. We should do maybe something like how the credits work: A plug-and-play thing for the basic use case, and you just gotta roll your own custom stuff if you want something fancy.

**Comment by MondayHopscotch** [comment]
I stripped out the existing flixel-ui stuff and just made it a simple title screen with a single "press start" thing on it. Flixel-ui is dumb. Ideally, we'd start using something like [haxe-ui](https://github.com/haxeui/haxeui-flixel), but I've had some issues getting that to cooperate, and I don't know that it has good controller support yet. Last time I tried it was a good while ago.

**Comment by Kenoshen** [comment]
Ya, I think your comment on making it plug-and-play is a good idea.  And given that 99% we have just a Start and Credits button, I feel like we could probably just roll our own and make it work with controllers, keyboard, mouse.  Then provide some easy-to-understand config that allows you to change the values around a bit (like title image, button images, button text, etc).  I would like to get animated title screens too.  So maybe we hook into the animation stuff a bit too.

**Comment by MondayHopscotch** [comment]
Maybe I missed something? The title screen as it was in the template as you used it should have had a "Start" and "Credits" button that work with mouse/controller. The title image was also an image that had the path of where it was so that you know what to change.

It didn't support animated images, nor different images for the buttons, but it should have been usable outside of that customization.

The new one is now _no_ buttons, it just has an image saying "Press Start" and just waits for the player to hit start. It's much simpler than the old one and avoids some of the weird xml stuff that we used to use.
